pkg/yurtappmanager/controller/uniteddeployment/adapter/deployment_adapter.go

```diff
@@ -136,6 +137,32 @@ func (a *DeploymentAdapter) ApplyPoolTemplate(ud *alpha1.UnitedDeployment, poolN
 	set.Spec.ProgressDeadlineSeconds = ud.Spec.WorkloadTemplate.DeploymentTemplate.Spec.ProgressDeadlineSeconds
 
 	attachNodeAffinityAndTolerations(&set.Spec.Template.Spec, poolConfig)
+
+	if poolConfig.Patches == nil {
+		// If No Patches, Must Set patches annotation
+		set.Annotations[alpha1.AnnotationPatchesKey]=""
+		klog.Infof("Deployment[%s/%s-] has no patches, do not need strategicmerge", set.Namespace,
+			set.GenerateName)
+		return nil
+	}
+
+	patched := &appsv1.Deployment{}
+	if err := StrategicMergeByPatches(set, poolConfig.Patches, patched); err != nil {
+		klog.Errorf("Deployment[%s/%s-] strategic merge by patch %s error %v", set.Namespace,
+			set.GenerateName, string(poolConfig.Patches.Raw), err)
+		return err
+	}
+	klog.Infof("Deployment [%s/%s-] has patches configure successfully:%v", set.Namespace,
+		set.GenerateName, string(poolConfig.Patches.Raw))
+
+	*set = *patched
```

Comment:
I think you need to use DeepCopy method here.

Reply:
done